lib/versioning/hashicorp/index.ts

```diff
@@ -49,14 +49,14 @@ function getNewValue({
         replaceValue = `$<prefix>${npm.getMinor(newVersion)}$<suffix>`;
       }
       return currentValue.replace(
-        /(?<prefix>~>\s*0\.)\d+(?<suffix>.*)$/, // TODO #12070
+        regEx(`(?<prefix>~>\\s*0\\.)\\d+(?<suffix>.*)$`), // TODO #12071
         replaceValue
       );
     }
     // handle special ~> 1.2 case
     if (regEx(/(~>\s*)\d+\.\d+$/).test(currentValue)) {
       return currentValue.replace(
-        /(?<prefix>~>\s*)\d+\.\d+$/, // TODO #12070
+        regEx(`(?<prefix>~>\\s*)\\d+\\.\\d+$`), // TODO #12071
```

Comment:
```suggestion
        regEx(`(?<prefix>~>\\s*)\\d+\\.\\d+$`),
```

I think #12071 no longer applies, as we're cache the regex instances.

https://github.com/renovatebot/renovate/blob/32d397348996ebe580a9eda4a0506f56728a680a/lib/util/regex.ts#L22-L39

So i think we can remove that todo from all `regEx` calls

Reply:
I think we should ideally still move them to the module scope, or do you think there's no benefit?

Reply:
It would only make a little difference, as the function call will be saved. I don't think this will make a big difference.